harness/determined/pytorch/_pytorch_context.py

```diff
@@ -57,4 +82,224 @@ def get_lr_scheduler(self) -> Optional[pytorch.LRScheduler]:
             * ``optimizer()``
             * ``create_lr_scheduler()``
         """
-        return self.lr_scheduler
+        # TODO(DET-3267): deprecate this when releasing pytorch flexible primitives.
+        check.lt_eq(len(self.lr_schedulers), 1)
+        if len(self.lr_schedulers) == 1:
+            return self.lr_schedulers[0]
+        return None
+
+    def _load_trial(self, trial: "pytorch.PyTorchTrial") -> None:
```

Comment:
question: once we deprecate the old interface this function is going to go away?

Reply:
yes

Reply:
And moved this function to `PytorchTrialController`. Maybe it would be clearer